subgraphs/beefy-finance/src/mappings/protocol.ts

```diff
@@ -0,0 +1,144 @@
+import { BigDecimal, dataSource, ethereum } from "@graphprotocol/graph-ts";
+import { Vault, YieldAggregator } from "../../generated/schema";
+import {
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  PROTOCOL_ID,
+} from "../prices/common/constants";
+import {
+  createFirstDailyFinancialSnapshot,
+  getUniqueUsers,
+  updateDailyFinancialSnapshot,
+  updateUsageMetricsDailySnapshot,
+  updateUsageMetricsHourlySnapshot,
+} from "../utils/metrics";
+import { getVaultDailyRevenues, updateVaultAndSave } from "./vault";
+
+export function createBeefyFinance(
+  vaultId: string,
+  block: ethereum.Block
+): YieldAggregator {
+  const beefy = new YieldAggregator(PROTOCOL_ID);
+  beefy.name = "Beefy Finance";
+  beefy.slug = "beefy-finance";
+  beefy.schemaVersion = "1.2.1";
+  beefy.subgraphVersion = "1.0.2";
+  beefy.methodologyVersion = "1.1.0";
+  beefy.network = dataSource
+    .network()
+    .toUpperCase()
+    .replace("-", "_");
+  beefy.type = "YIELD";
+  beefy.totalValueLockedUSD = BIGDECIMAL_ZERO;
+  beefy.protocolControlledValueUSD = BIGDECIMAL_ZERO;
+  beefy.cumulativeSupplySideRevenueUSD = BIGDECIMAL_ZERO; //todo
+  beefy.cumulativeProtocolSideRevenueUSD = BIGDECIMAL_ZERO; //todo
+  beefy.cumulativeTotalRevenueUSD = BIGDECIMAL_ZERO; //todo
+  beefy.cumulativeUniqueUsers = BIGINT_ZERO;
+  beefy.vaults = [vaultId];
+  beefy.dailyUsageMetrics = [updateUsageMetricsDailySnapshot(block, beefy).id];
+  beefy.hourlyUsageMetrics = [
+    updateUsageMetricsHourlySnapshot(block, beefy).id,
+  ];
+  beefy.financialMetrics = [createFirstDailyFinancialSnapshot(block, beefy).id];
```

Comment:
you don't need to update referenced entities. they are handled automatically by `theGraph`.

Reply:
The update function is here also to create a new snapshot if it has not been created yet.
Do you want me to update the function's name to make it clear?

Reply:
I meant you don't need to update the `many` side of `derivedFrom` fields. The 'many' side is automatically derived. So you don't need to update `dailyUsageMetrics`, `hourlyUsageMetrics`,  `financialMetrics` here. [Ref](https://thegraph.com/docs/en/developer/create-subgraph-hosted/#:~:text=Reverse%20Lookups%23,as%20is%20practical.)

Reply:
ok, now I fully understood, thanks for the clarification